settings.php

```diff
@@ -57,14 +58,14 @@
             );
             $settings->add($setting);
 
-            // Image.
-            $setting = new admin_setting_configtext('block_news/image'.$i,
+            // Adding a file upload element for an image.
+            $settings->add(new admin_setting_configstoredfile(
+                'block_news/image'.$i,
                 get_string('image', 'block_news'),
                 get_string('image_help', 'block_news'),
-                $default,
-                PARAM_RAW
-            );
-            $settings->add($setting);
+                'block_news',
+                $i
+            ));
```

Comment:
Seems to be missing $settings->add($setting);

Reply:
No need to as I add it directly to $settings.